### ISSUE TITLE
Certain unicode characters are not rendered

### ISSUE THREAD

**Issue by alexpyattaev** [lapce/lapce#2614]
## Lapce Version

The Lapce version 0.2.8 

## System information

Operating system used: Arch linux, x86_64

## Describe the bug
The rendering of certain multibyte unicode characters does not happen. This does not seem to be font related, as changing font does not change the rendering of the character. It seems that if character has >2 bytes it does not get rendered correctly.

Specific example:
``` rust
    let str_content = "𒀕bcdefghijk"; // the 4-byte unicode character comes first in this string
```

## Additional information
Here is what Lapce shows:
![image](https://github.com/lapce/lapce/assets/5932526/1ecb543a-7a21-4a5a-8011-ca776cc4f2c4)

The actual correctly rendered string is:
![image](https://github.com/lapce/lapce/assets/5932526/25d6f4ce-71ad-42ec-aca8-62345883e7d6)

The saddest part is that there is no rendering at all, so you can assume the string is empty while it is not, which may cause users to lose sanity really quickly.

**Comment by dzhou121** [comment]
Can you try the nightly version? We changed the text layout engine.

**Comment by alexpyattaev** [comment]
Yep. Its different. But still wrong.

![image](https://github.com/lapce/lapce/assets/5932526/d5442a01-6280-464f-abc8-4e943db52b58)

I think the main issue is that once it gives up figuring up what the character does, it just assumes it takes no space. Or it can not figure how much space it takes and assumes 0?

**Comment by dzhou121** [comment]
I think we'll first fix the issue of not taking space if a character is not found in the font. Then we'll fix the issue of displaying the character. Apparently there should be some font that's got this character but we failed to fall back to it.

**Comment by alexpyattaev** [comment]
Yes, that sounds like a good plan! Having some squiggly nonsense would be way better than nothing at all! Thank you!
